core/native/src/main/scala/bobcats/HashPlatform.scala

```diff
@@ -16,27 +16,82 @@
 
 package bobcats
 
-import cats.effect.kernel.Async
+import scala.scalanative.annotation.alwaysinline
+import cats.effect.kernel.{Async, Resource, Sync}
 import scodec.bits.ByteVector
 import scalanative.unsafe._
 import scalanative.unsigned._
 import openssl._
 import openssl.err._
 import openssl.evp._
 
+private final class NativeEvpDigest[F[_]](val ctx: Ptr[EVP_MD_CTX], digest: Ptr[EVP_MD])(
+    implicit F: Sync[F])
+    extends UnsealedDigest[F] {
+
+  override def update(data: ByteVector): F[Unit] = F.delay {
+
+    val d = data.toArrayUnsafe
+    val len = d.length
+
+    if (EVP_DigestUpdate(ctx, if (len == 0) null else d.at(0), d.length.toULong) != 1) {
+      throw Error("EVP_DigestUpdate", ERR_get_error())
+    }
+  }
+
+  override val reset = F.delay {
+    if (EVP_MD_CTX_reset(ctx) != 1) {
+      throw Error("EVP_MD_Ctx_reset", ERR_get_error())
+    }
+    if (EVP_DigestInit_ex(ctx, digest, null) != 1) {
+      throw Error("EVP_DigestInit_ex", ERR_get_error())
+    }
+  }
+
+  override def get: F[ByteVector] = F.delay {
+    val md = stackalloc[CUnsignedChar](EVP_MAX_MD_SIZE)
+    val s = stackalloc[CInt]()
+
+    if (EVP_DigestFinal_ex(ctx, md, s) != 1) {
+      throw Error("EVP_DigestFinal_ex", ERR_get_error())
+    }
+    ByteVector.fromPtr(md.asInstanceOf[Ptr[Byte]], s(0).toLong)
+  }
+
+  def free: F[Unit] = F.delay(EVP_MD_CTX_free(ctx))
+
+}
+
 private[bobcats] trait HashCompanionPlatform {
-  implicit def forAsync[F[_]](implicit F: Async[F]): Hash[F] =
+  implicit def forAsync[F[_]](implicit F: Async[F]): Hash[F] = forSync
```

Comment:
Btw: I now believe offering implicits in exchange for `Sync`/`Async` is an anti-pattern. See:
- https://github.com/typelevel/fs2/pull/3179